modules/UI/videolayout/ConnectionIndicator.js

```diff
@@ -351,23 +357,30 @@ ConnectionIndicator.prototype.remove = function() {
 };
 
 /**
- * Updates the UI which displays warning about user's connectivity problems.
+ * Updates the UI which displays or not a warning about user's connectivity
+ * problems.
  *
- * @param {boolean} isActive true if the connection is working fine or false if
- * the user is having connectivity issues.
+ * @param {ParticipantConnectionStatus} connectionStatus
  */
 ConnectionIndicator.prototype.updateConnectionStatusIndicator
-    = function (isActive) {
-        this.isConnectionActive = isActive;
-        if (this.isConnectionActive) {
-            $(this.interruptedIndicator).hide();
-            $(this.emptyIcon).show();
-            $(this.fullIcon).show();
-        } else {
-            $(this.interruptedIndicator).show();
-            $(this.emptyIcon).hide();
-            $(this.fullIcon).hide();
-        }
+= function (connectionStatus) {
+    this.connectionStatus = connectionStatus;
+    if (connectionStatus === ParticipantConnectionStatus.INTERRUPTED) {
+        $(this.interruptedIndicator).show();
+        $(this.emptyIcon).hide();
+        $(this.fullIcon).hide();
+        $(this.ninjaIndicator).hide();
+    } else if (connectionStatus === ParticipantConnectionStatus.INACTIVE) {
+        $(this.interruptedIndicator).hide();
+        $(this.emptyIcon).hide();
+        $(this.fullIcon).hide();
+        $(this.ninjaIndicator).show();
+    } else {
+        $(this.interruptedIndicator).hide();
+        $(this.emptyIcon).show();
+        $(this.fullIcon).show();
+        $(this.ninjaIndicator).hide();
+    }
```

Comment:
Hmm I've just tried that, but it doesn't look quite readable. It' hard to figure out when each icon is displayed, at least for me, WDYT ?

<img width="811" alt="screen shot 2017-06-05 at 14 21 24" src="https://cloud.githubusercontent.com/assets/2965063/26799225/534a175a-49fa-11e7-9ce7-9b1eb19e2b15.png">